### ISSUE TITLE
Prerelease version calculation doesn't consider commits

### ISSUE THREAD

**Issue by mpsanchis** [jscutlery/semver#688]
Hi all,

I would like to point out that the version calculation when running semver with  `--releaseAs prerelease` is not what I expect. I have gone through the issues, and found that [#459](https://github.com/jscutlery/semver/issues/459), [#601](https://github.com/jscutlery/semver/issues/601), [#679](https://github.com/jscutlery/semver/issues/679) and [#428](https://github.com/jscutlery/semver/issues/428) point to the same direction. 

These issues are mentioned in discussion #385, but I think they were never addresed. [This comment](https://github.com/jscutlery/semver/discussions/385#discussioncomment-1663718) by @yjaaidi was giving a very precise "Expected Behavior" scenario. 

In a nutshell, when running semver with `--releaseAs prerelease`, we would expect the version calculator to consider the commit types, and decide what number to bump (major, minor, patch or prerelease) based on both the last tagged commit and all commits since that one.

Is there any known workaround, or a way to do this properly? I am currently running `prerelease` in my CI, so forcing the bump (premajor/preminor/prepatch) is not an option, since I run the same NX command.

Thank you

**Comment by yjaaidi** [comment]
Thanks @mpsanchis for your feedback on this.
This is in fact something that we didn't discuss yet concerning semver 3 and it's good that you raised the issue.

We have to include this in the discussion. If you have any suggestions on how you'd like to see this implemented with the new strategy system, then go ahead 😊

**Comment by eXigentCoder** [comment]
In the same boat here, would really love it if the `prerelease` could consider commits and generate a version with the `preId` based off of that!

Example below for clarity:

```
Start version commit & tag: 0.0.1 [Branch: main]
feat: [releaseAs: not set] <-- package@0.1.0 [Branch: main]
fix:  [releaseAs: not set] <-- package@0.1.1 [Branch: main]
feat: [releaseAs: prerelease, preId: beta] <-- package@0.2.0-beta.0 [Branch: dev]
feat: [releaseAs: prerelease, preId: beta] <-- package@0.2.0-beta.1 [Branch: dev]
fix: [releaseAs: prerelease, preId: beta] <-- package@0.2.0-beta.2 [Branch: dev]
merge to main branch --> [releaseAs: not set] <-- package@0.2.0
```

Hopefully I got it right and this helps?

**Comment by sunglam** [comment]
@yjaaidi Hi, did you get any solution for this issue? Prerelease doesn't care about commits and commit types.
